parser/Parser/Reader.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+#include <cstdio>
+#include <string>
+
+/// C++ wrapper for reading files
+class Reader {
+    FILE* f;
+public:
+    /// takes ownership of FILE* resource(!)
+    Reader(FILE* f) : f(f) {}
+    ~Reader() {
+        fclose(f);
+    }
+
+    std::string readUntilChar(char end) {
+        bool found = false;
+        std::string result = std::string();
```

Comment:
This calls the constructor :)

I guess you have recently coded in Java :D (In Java, everything is a reference/pointer, and initialized to null. In C/C++, local variables are initialized by constructors and they are allocated enough space, even without explicit constructor call.)

```suggestion
        std::string result;
```